docs/proposals/container-runtime-interface/capabilities.md

```diff
@@ -0,0 +1,202 @@
+- From a security point of view, an explicit capability set which does not rely on undefined defaults is preferable for auditing purposes.
+
+## Solution
+
+The proposed solution will:
+- Keep the Kubernetes API backwards compatible.
+- Add a new `RequestedSet` field to the `PodSpec.SecurityContext.Capabilities` struct.
+- Add a new `DefaultCapabilities` field to the `PodSecurityPolicySpec` struct.
+- Define a default capability whitelist inside the following Kubernetes components:
+  1. the *API server*: Capability defaults are being applied to `PodSpec.SecurityContext.Capabilities.RequestedSet` either through the `PodSecurityPolicySpec` or, if admission control is disabled, using a default capability whitelist.
+  2. the *Kubelet*: If the `PodSpec.SecurityContext.Capabilities.RequestedSet` does not specify any capabilities, the kubelet applies capability defaults to retain backwards compatibility.
+- Make CRI expose the whitelist based capability interface, remove the existing `add_capabilities`, and `drop_capabilities` fields, and put the burden on the runtime to implement that.
+- Make the final requested capability set **visible** through the container spec, for example:
+
+```shell
+$ kubectl get pod -o yaml
+apiVersion: v1
+items:
+- apiVersion: v1
+  kind: Pod
+  metadata:
+  ...
+  spec:
+    containers:
+    - command:
+      - sleep
+      - "1000"
+      image: busybox
+      imagePullPolicy: Always
+      name: my-busybox
+      securityContext:
+        capabilities:
+          requestedSet:
+          - SETPCAP, MKNOD, AUDIT_WRITE, ...
+          add:
+          - NET_ADMIN
+          drop:
+          - MKNOD
+      ...
+  status:
+    containerStatuses:
+    - image: busybox
+      name: my-busybox
+      state:
+        running:
+          startedAt: 2016-09-27T22:33:38Z
+    ...
+```
+
+## Struct changes
+
+The `PodSecurityPolicySpec` in `pkg/apis/extensions/types.go` gets a new `DefaultCapabilities` field. The existing fields `DefaultAddCapabilities`, and `RequiredDropCapabilities` will be marked as deprecated.
+
+```
+type PodSecurityPolicySpec struct {
+...
+        // DefaultAddCapabilities is the default set of capabilities that will be added to the container
+        // unless the pod spec specifically drops the capability or the pod spec uses a requested-set
+        // You may not list a capability in both DefaultAddCapabilities and RequiredDropCapabilities.
+        // Note: this field is deprecated in favor of the DefaultCapabilities field.
+        DefaultAddCapabilities []api.Capability `json:"defaultAddCapabilities,omitempty"`
+
+        // RequiredDropCapabilities are the capabilities that will be dropped from the container. These
+        // are required to be dropped and cannot be added or present in the requested-set.
+        // Note: this field is deprecated in favor of the DefaultCapabilities field.
```

Comment:
I think I may be missing something. Can `DefaultCapabilities` actually express something of equivalent power?

IIUC now, this field could be used to ensure a pod never has `CAP_NETADMIN` even if it was added in the podspec, but post this change I don't see how the same goal would be accomplished.